### PR TITLE
Fixes error scope tests for Linux/Intel

### PR DESCRIPTION
**Pull request by lokokung**
Fixes the error scope tests on Linux/Intel once the associated [fix](https://dawn-review.googlesource.com/c/dawn/+/159662) lands.

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/api/validation/error_scope.spec.ts [hunk 1]
```diff
@@ -146,7 +155,7 @@ Tests that popping an empty error scope stack should reject.
   )
   .fn(t => {
     const promise = t.device.popErrorScope();
-    t.shouldReject('OperationError', promise);
```
Comment (kainino0x) [inline src/webgpu/api/validation/error_scope.spec.ts:149]: this is fine since we already test popErrorScope has stacks elsewhere, but why is this needed?

File: src/webgpu/api/validation/error_scope.spec.ts [hunk 2]
```diff
@@ -26,7 +26,16 @@ class ErrorScopeTests extends Fixture {
...
     const adapter = await gpu.requestAdapter();
     assert(adapter !== null);
-    const device = await adapter.requestDevice();
+
+    // We need to max out the adapter limits related to texture dimensions to more reliably cause an
+    // OOM error when asked for it, so set that on the device now.
+    const device = this.trackForCleanup(
+      await adapter.requestDevice({
+        requiredLimits: {
+          ['maxTextureDimension2D']: adapter.limits.maxTextureDimension2D,
```
Comment (kainino0x) [inline src/webgpu/api/validation/error_scope.spec.ts:35]: nit
```suggestion
          maxTextureDimension2D: adapter.limits.maxTextureDimension2D,
```

**Review by kainino0x — APPROVED** [review]

File: src/webgpu/api/validation/error_scope.spec.ts (hunk 1 above)
Comment (lokokung) [reply src/webgpu/api/validation/error_scope.spec.ts:149]: I'm actually not entirely sure? Maybe I had an outdated version of chromium that didn't plumb the stack through, but when I was running the tests, without this they were all failing with the missing stack error... I can't easily rebase/update my chromium atm so I just added these suppressions for the time being. If we are confident it should work without them I can remove them also.

Comment (lokokung) [reply src/webgpu/api/validation/error_scope.spec.ts:149]: Actually, since I'm OOO I will just land this for now. If it turns out we don't need the allow `allowMissingStacks` we can fix them in a follow up!

Comment (kainino0x) [reply src/webgpu/api/validation/error_scope.spec.ts:149]: You didn't need to check back on this while OOO! I was assuming it was just an old chromium build, and I was going to try removing these changes, but they really don't matter, so no need to follow up.
